src/megaclient.cpp

```diff
@@ -10955,6 +10955,72 @@ bool MegaClient::syncup(LocalNode* l, dstime* nds)
                 // file on both sides - do not overwrite if local version older or identical
                 if (ll->type == FILENODE)
                 {
+                    if (ll->node != rit->second)
+                    {
+                        ll->sync->statecacheadd(ll);
+                    }
+                    ll->setnode(rit->second);
+
+                    // check if file is likely to be identical
+                    if (*ll == *(FileFingerprint*)rit->second)
```

Comment:
May it happen that `!rit->second->isValid` at this point? It's a condition that used to be and now it's not ported to this check.
Additionally, there use to be a check for the size that has not been ported either:
```if (ll->size == rit->second->size)```

Reply:
Yes, it could happen. The value of `isValid` is taken into account in the overloaded operator `==` of `FileFingerprint`, resulting on a check similar to the one that has been replaced, but taking also into account that the modification times can be slightly different due to its resolution on some filesystems like FAT.